File: jwt/src/main/java/com/example/jwt/Entity/Declarant.java
package com.example.jwt.Entity;

import javax.persistence.*;
import java.sql.Time;
import java.time.LocalTime;
import java.util.Date;
import java.util.Set;

@Entity
public class Declarant extends Personne{

    private String degreParental;

    @OneToMany(mappedBy="declarant")
    private Set<Naissance> naissances;

    @OneToMany(mappedBy="declarant")
    private Set<Mariage> mariages;

    @OneToMany(mappedBy="declarant")
    private Set<Deces> deces;

    public Declarant(Long id,
                     String prenom,
                     String nom,
                     String profession,
                     String CNI,
                     String sexe,
                     String adresse,
                     String lieuNaissance,
                     Date dateNaissance,
                     LocalTime heureNaissance,
                     String degreParental) {
        super(id, prenom, nom, profession, CNI, sexe, adresse, lieuNaissance, dateNaissance, heureNaissance);
        this.degreParental = degreParental;
    }

    public Declarant() {

    }

}

File: frontmairie/src/app/dashboard/home/Declarant.ts
import {Time} from "@angular/common";

export class Declarant{
  id: number | undefined;
  prenom: string | undefined;
  nom: string | undefined;
  profession: string | undefined;
  CNI: string | undefined;
  sexe: string | undefined;
  adresse: string | undefined;
  lieuNaissance: string | undefined;
  dateNaissance: Date | undefined;
  heureNaissance: Time | undefined;
  degreParental: string | undefined;
}
